### PR TITLE
Docs/sdk 3968/new arch docs

### PR DESCRIPTION
**Pull request by Anush-Shand**
Adds doc changes for RN 300

- Changelog
- Explayer migration
- Legacy version updates
- Semi breaking change related to new arch `ClevertapPushNotificationClicked` and other app startup callbacks
- Note regarding new arch
- RN version support

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,8 +11,8 @@ Version 3.0.0 *(Todo 2024)*
   * Deprecates `CleverTapModule.setInitialUri()` in favour of `CleverTapRnAPI.setInitialUri()`. Refer to [step 4](docs/integration.md#step4)
 
 * **[iOS Platform]**
-  * Supports [CleverTap iOS SDK v7.0.1](https://github.com/CleverTap/clevertap-android-sdk/blob/master/docs/CTCORECHANGELOG.md#version-701-september-2-2024).
-  * Todo
+  * Supports [CleverTap iOS SDK v7.0.1](https://github.com/CleverTap/clevertap-ios-sdk/blob/master/CHANGELOG.md#version-701-august-22-2024).
+  * Fixes a bug where some user properties were being sent with an incorrect prefix.
```
Comment (Anush-Shand) [inline CHANGELOG.md:15]: This bug was only added in v7.0.0 which was never released for RN? @akashvercetti

**Review by akashvercetti — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,35 @@
...
+  * Supports [CleverTap iOS SDK v7.0.1](https://github.com/CleverTap/clevertap-ios-sdk/blob/master/CHANGELOG.md#version-701-august-22-2024).
+
+* **[Android and iOS Platform]**
+  * Adds support for triggering InApps based on user attribute changes.
+  * Migrates the bridge to a backwards-compatible [New Architecture Turbo Module](docs/integration.md).
+    * The CleverTap ReactNative SDK continues to be compatible with both the Old and the New Architecture.
+
+**Breaking Changes**
+* **[Android Platform]**
+  * If you're using a custom `Application` class, ensure that it extends the `CleverTapApplication` or calls `CleverTapRnAPI.initReactNativeIntegration(this);` to enable the functionality of `ClevertapPushNotificationClicked` and few other callbacks from killed state. Refer to [step 3](docs/integration.md#step3a)
```
Comment (vasct) [inline CHANGELOG.md:23]: I think the RN apps must have a custom `Application` class that implements ReactApplication.

File: docs/integration.md [hunk 3]
```diff
@@ -42,24 +47,67 @@ Note: Need to use **@import CleverTapSDK;** instead of **#import <CleverTap-iOS-
...
-    // add onCreate() override
-    @Override
-    public void onCreate() {
-	    // Register the CleverTap ActivityLifecycleCallback; before calling super
-        ActivityLifecycleCallback.register(this);	
-        super.onCreate();
-        // Init CleverTapSDK React Native integration
-        CleverTapRnAPI.initReactNativeIntegration(this);
-    }
+        public class MainApplication implements ActivityLifecycleCallbacks, ReactApplication {
```
Comment (vasct) [inline docs/integration.md:64]: Is it required for this class to implement ActivityLifecycleCallbacks ?

**Review by vasct — APPROVED** [review]

**Review by CTLalit — COMMENTED** [review]
There is a suggestion in comments (not from me).

File: CHANGELOG.md (hunk 2 above)
Comment (Anush-Shand) [reply CHANGELOG.md]: Updated. Thanks

File: docs/integration.md (hunk 3 above)
Comment (Anush-Shand) [reply docs/integration.md]: Updated thanks

**Pull request merged**